src/users/dto/update-user.dto.ts

```diff
@@ -0,0 +1,19 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateUserDto } from './create-user.dto';
+import { IsAlphanumeric, IsEmail, IsEnum, IsOptional } from 'class-validator';
+import { Role } from '../../../common/enums/users.role';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UpdateUserDto extends PartialType(CreateUserDto) {
```

Comment:
Do you need to define fields if you use `extends PartialType(CreateUserDto)` ? If I am not mistaken the fields are already optional when you use `PartialType`.
 Please try as follows:
`export class UpdateUserDto extends PartialType(CreateUserDto) {}`